src/rtp_table.cpp

```diff
@@ -1044,6 +1042,42 @@ const char* rtp_table_2k[][5] = {
 	{nullptr, nullptr, nullptr, nullptr, nullptr}
 };
 
+const char* rtp_table_2k_categories[15] = {
+	"backdrop",
+	"battle",
+	"charset",
+	"chipset",
+	"faceset",
+	"gameover",
+	"monster",
+	"movie",
+	"music",
+	"panorama",
+	"picture",
+	"sound",
+	"system",
+	"title",
+	nullptr
+};
+
+int rtp_table_2k_categories_idx[15] = {
+	0,
+	39,
+	69,
+	282,
+	418,
+	521,
+	522,
+	613,
+	614,
+	741,
+	755,
+	756,
+	978,
+	1002,
+	1006
```

Comment:
What you're doing makes sense from a performance standpoint, but maintaining these tables is very brittle. Possible to write any unit test or `static_assert()` to ensure correctness here?

Reply:
the numbers are calculated by the bundled Python script, will try to add a check to ensure the mapping is up-to-date.